cmd/livepeer/livepeer.go

```diff
@@ -553,23 +554,23 @@ func main() {
 				glog.Errorf("Error setting up PM recipient: %v", err)
 				return
 			}
+		}
 
-			// Only initialize rounds and call reward if the node is using an on-chain registered orchestrator address
-			// Assume that the node is using an on-chain registered orchestrator address if -ethOrchAddr is not specified or
-			// if the ticket recipient is the node's address
-			if *ethOrchAddr == "" || n.Eth.Account().Address == recipientAddr {
-				// Create round iniitializer to automatically initialize new rounds
-				if *initializeRound {
-					initializer := eth.NewRoundInitializer(n.Eth, n.Database, timeWatcher, blockPollingTime)
-					go initializer.Start()
-					defer initializer.Stop()
-				}
-
-				// Create reward service to claim/distribute inflationary rewards every round
-				rs := eventservices.NewRewardService(n.Eth, blockPollingTime)
-				rs.Start(ctx)
-				defer rs.Stop()
+		// Only initialize rounds and call reward if the node is using an on-chain registered orchestrator address
+		// Assume that the node is using an on-chain registered orchestrator address if -ethOrchAddr is not specified or
+		// if the ticket recipient is the node's address
+		if *ethOrchAddr == "" || n.Eth.Account().Address == recipientAddr {
```

Comment:
Since this conditional is outside of the `if *orchestrator { ... }` block, the code below will be executed when a node runs with `-broadcaster`.

This could be addressed with an updated conditional such as the following:

```golang
if (*orchestrator && *ethOrchAddr == "") || (n.Eth.Account().Address == *ethOrchAddr && *ethOrchAddr != "") { ... }
```

The first part of the conditional preserves existing reward service behavior for orchestrators - if the orchestrator is run without `-ethOrchAddr` (meaning it is not a part of a multi-O setup) then the reward service is started.

The second part of the conditional ensures that following code is only executed if the `-ethOrchAddr` is specified and the address is equal to the node's address (the problem with the current conditional is that `n.Eth.Account().Address` will be equal to `recipientAddr` whenever `-ethOrchAddr` is not specified which is the case when a user runs a broadcaster).

Reply:
Actually I realized a more straightforward way to handle this could be the following:

```golang
// If the node address is an on-chain registered address, start the reward service
t, err := n.Eth.GetTranscoder(n.Eth.Account().Address)
if err != nil {
    ...
}
if t.Status == "Registered" {
    // Start reward service
}

// The node will only initialize rounds if it in the upcoming active set for the round
if *initializeRound {
    // Start round initializer
}
```

These changes would result in the following:
- If `livepeer` is run with an on-chain registered address, the reward service is started (but the node will only call reward if it is active in the current round).
- If `livepeer` is run with `-initializeRound`, the round initializer is started (but the node will only initialize rounds if it is in the upcoming active set for the round).